packages/odyssey-react-mui/src/labs/DataView/DataView.tsx

```diff
@@ -262,49 +278,62 @@ const DataView = ({
     return;
   }, [noResultsPlaceholder, t, isEmpty, isNoResults, emptyPlaceholder]);
 
-  const additionalActions = useMemo(
-    () => (
-      <>
-        {currentLayout === "table" && tableLayoutOptions && (
-          <TableSettings
-            setTableState={setTableState}
-            tableLayoutOptions={tableLayoutOptions}
-            tableState={tableState}
-          />
-        )}
-
-        {availableLayouts.length > 1 && (
-          <LayoutSwitcher
-            availableLayouts={availableLayouts}
-            currentLayout={currentLayout}
-            setCurrentLayout={setCurrentLayout}
-          />
-        )}
-
-        {additionalActionButton}
-
-        {additionalActionMenuItems && (
-          <MenuButton
-            endIcon={<MoreIcon />}
-            ariaLabel={t("table.moreactions.arialabel")}
-            buttonVariant="secondary"
-            menuAlignment="right"
-          >
-            {additionalActionMenuItems}
-          </MenuButton>
-        )}
-      </>
-    ),
-    [
-      currentLayout,
-      tableLayoutOptions,
-      tableState,
-      availableLayouts,
-      additionalActionButton,
-      additionalActionMenuItems,
-      t,
-    ],
-  );
+  const additionalActions = useMemo(() => {
+    return (
+      (metaText ||
```

Comment:
We can make this an implicit return right?

But even then, there's a lot going on in here.

Is there a reason this isn't just another component? React v19 would make this separate `useMemo` hook, but we don't need to do it here. It's only boolean checks. This just seems unnecessarily complicated.

Reply:
I like the idea of making this another component, but I don't have time to tackle that right now. Would you be comfortable with us shipping as-is and then doing a fast-follow PR to break this out into a new component?

Reply:
Yes because it's more of a code styling thing. It doesn't affect consumers.